### PR TITLE
Allow SDK to provide its own custom subgraph

### PR DESCRIPTION
**Pull request by anxolin**
The goal of this PR is to let an SDK client to define its own subgraph API URLs.

They will be optional parameters, and it defaults to the current URLs.

The reason I'm doing this and creating a version ASAP is to test Satsuma in the Explorer. 

Now clients will be able to instantiate the SDK like this:
```js
const cowSdk = new CowSdk(100, {
  subgraphBaseUrls: {
    1: 'https://api.thegraph.com/subgraphs/name/cowprotocol/cow',
    5: 'https://api.thegraph.com/subgraphs/name/cowprotocol/cow-goerli',
    100: 'https://api.thegraph.com/subgraphs/name/cowprotocol/cow-gc',
  },
})
```

## Test

* [satsuma] Goerli test: https://github.com/cowprotocol/explorer/pull/368
* [Satsuma] Mainnet + Gnosis Chain test: https://github.com/cowprotocol/explorer/pull/369

### CODE REVIEW

**Review by elena-zh — COMMENTED** [review]
Hey @anxolin , I have done Explorer tests on 3 networks, and everything works smoothly there. Only one thing that I noticed is that this console error when the app is loaded: https://reactjs.org/docs/error-decoder.html/?invariant=321
image

**Review by elena-zh — APPROVED** [review]

File: src/api/cow-subgraph/index.ts [hunk 1]
```diff
@@ -7,19 +7,27 @@ import { SupportedChainId as ChainId } from '../../constants/chains'
 import { LastDaysVolumeQuery, LastHoursVolumeQuery, TotalsQuery } from './graphql'
 import { LAST_DAYS_VOLUME_QUERY, LAST_HOURS_VOLUME_QUERY, TOTALS_QUERY } from './queries'
 
-export function getSubgraphUrl(env: Env): Partial<Record<ChainId, string>> {
+const DEFAULT_STAGING_URLS = {
+  [ChainId.GOERLI]: 'https://subgraph.satsuma-prod.com/94b7bd7c35c5/cow/cow-goerli/api',
```
Comment (alfetopito) [inline src/api/cow-subgraph/index.ts:11]: Satsuma is the default now, only for Goerli?

**Review by alfetopito — APPROVED** [review]
Approved, only question regarding the new default values

**Comment by shoom3301** [comment]
https://github.com/cowprotocol/cow-sdk/blob/main/src/subgraph/api.ts#L26
@anxolin I've added `customEnvConfigs` for this case

File: src/api/cow-subgraph/index.ts (hunk 1 above)
Comment (alfetopito) [reply src/api/cow-subgraph/index.ts:11]: Shouldn't be the default anymore as we ended the trial period

**Pull request closed without merging**